tests/models/test_measurements.py

```diff
@@ -170,6 +170,57 @@ def test_measurements_cannot_be_saved_twice(ptf_datastore):
                 session.delete(m2)
                 session.commit()
 
+def test_threshold_flagging(ptf_datastore, measurer):
+
+    measurements = ptf_datastore.measurements
+    m = measurements[0]  # grab the first one as an example
+
+    m.provenance.parameters['thresholds']['negatives'] = 0.3
+    measurer.pars.deletion_thresholds['negatives'] = 0.5
+
+    m.disqualifier_scores['negatives'] = 0.1 # set a value that will pass both
+    assert measurer.compare_measurement_to_thresholds(m) == "ok"
+
+    m.disqualifier_scores['negatives'] = 0.4 # set a value that will fail one
+    assert measurer.compare_measurement_to_thresholds(m) == "bad"
+
+    m.disqualifier_scores['negatives'] = 0.6 # set a value that will fail both
+    assert measurer.compare_measurement_to_thresholds(m) == "delete"
+
+    # test what happens if we set deletion_thresholds to unspecified
+    #   This should use threshold values as deletion_threshold values
+    measurer.pars.deletion_thresholds = {}
+
+    m.disqualifier_scores['negatives'] = 0.1 # set a value that will pass
+    assert measurer.compare_measurement_to_thresholds(m) == "ok"
+
+    m.disqualifier_scores['negatives'] = 0.4 # set a value that will fail
+    assert measurer.compare_measurement_to_thresholds(m) == "delete"
+
+    # I'm not sure how setting the value to infinity would look in the config yaml
+    # (so I could use a similar value in the test), but if we want to test it then
+    # I'm open to a more proper value
+
+    # test what happens if we set deletion_thresholds to very large numbers
+    #   This should effectively turn off deletion and only return 'ok' or 'bad'
+    measurer.pars.deletion_thresholds['negatives'] = 1000
+    m.disqualifier_scores['negatives'] = 0.1 # set a value that will pass
+    assert measurer.compare_measurement_to_thresholds(m) == "ok"
+
+    m.disqualifier_scores['negatives'] = 0.4 # a value that would fail mark
+    assert measurer.compare_measurement_to_thresholds(m) == "bad"
+
+    m.disqualifier_scores['negatives'] = 0.9 # a value that would fail both (earlier)
+    assert measurer.compare_measurement_to_thresholds(m) == "bad"
+
+    # test deletion_threshold varies as a non-critical parameter
```

Comment:
I think this section below might be a bad, trivial test of the non-critical parameter. Perhaps the proper way would be to actually use `measurer.run()` with the deletion dictionary set to two different values and ensure the provenances of resulting measurements objects come out the same. That might deserve moving to a separate test, if we think its a test worth writing.

Reply:
I added what I think may be a better test below for review.